.github/workflows/esbuild-publish.yml

```diff
@@ -0,0 +1,48 @@
+name: esbuild Compilation & npm Publish Workflow
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Set the branch to use for automation tests'
+        type: string
+        required: false
+        default: 'main'
+      nodeVersion:
+        description: version of node to use.  It's better to specify latest, lts/* or lts/-1 than to hardode numbers
+        type: string
+        default: lts/*
+        required: false
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.branch }}
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{ inputs.nodeVersion }}
+          registry-url: 'https://registry.npmjs.org'
+      - name: Install esbuild Dependencies
+        run: |
+          yarn add esbuild@^0.19.5 -D
+          yarn add esbuild-plugin-pino@^2.1.0 -D
+          yarn add npm-dts@^1.3.12 -D
+          yarn add esbuild-plugin-tsc@^0.4.0 -D
```

Comment:
you *could* just do 1 add, like

yarn add -D esbuild@^0.19.5 esbuild-plugin-pino@^2.1.0 etc.